packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -296,6 +347,31 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
       this._connections = new ec2.Connections({});
     }
 
+    if (props.runtime.family !== RuntimeFamily.NODEJS && props.artifactS3Encryption) {
+      throw new Error('Artifact encryption is only supported for Node.js runtime.');
```

Comment:
```suggestion
      throw new Error(`Artifact encryption is only supported for canaries that use Synthetics runtime version syn-nodejs-puppeteer-3.3 or later, got ${props.runtime.name}.`);
```
For consistency with CFN [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-artifactconfig.html).